pkgs/by-name/nu/nuget-to-nix/nuget-to-nix.sh

```diff
@@ -8,101 +9,59 @@ export PATH="@binPath@:$PATH"
+        echo "Null bytes found in file." >&2
+        exit 1
     fi
-    base_addresses[$index]="$base_address"
-done
-
-echo "{ fetchNuGet }: ["
-
-cd "$pkgs"
-for package in *; do
-  [[ -d "$package" ]] || continue
-  cd "$package"
-  for version in *; do
-    id=$(xmlstarlet sel -t -v /_:package/_:metadata/_:id "$version"/*.nuspec)
-
-    if grep -qxF "$id.$version.nupkg" "$excluded_list"; then
-      continue
+else
+    if ! jsonDeps=nuget-to-json "$@"; then
+        echo "nuget-to-json failed."
+        exit 1
     fi
+fi
 
-    # packages in the nix store should have an empty metadata file
-    used_source="$(jq -r 'if has("source") then .source else "" end' "$version"/.nupkg.metadata)"
-    found=false
-
-    if [[ -z "$used_source" || -d "$used_source" ]]; then
-      continue
-    fi
+# pkgs/by-name/az/azure-functions-core-tools/deps.json
+IFS='' readarray -d '' depsParts < <(jq 'map([.pname, .version, .sha256, .hash, .url]) | flatten[]' --raw-output0 <<<"$jsonDeps")
+index=0
 
-    for source in "${remote_sources[@]}"; do
-      url="${base_addresses[$source]}$package/$version/$package.$version.nupkg"
-      if [[ "$source" == "$used_source" ]]; then
-        hash="$(nix-hash --type sha256 --flat --sri "$version/$package.$version".nupkg)"
-        found=true
-        break
-      else
-        if hash=$(nix-prefetch-url "$url" 2>"$tmp"/error); then
-          hash="$(nix-hash --to-sri --type sha256 "$hash")"
-          # If multiple remote sources are enabled, nuget will try them all
-          # concurrently and use the one that responds first. We always use the
-          # first source that has the package.
-          echo "$package $version is available at $url, but was restored from $used_source" 1>&2
-          found=true
-          break
-        else
-          if ! grep -q 'HTTP error 404' "$tmp/error"; then
-            cat "$tmp/error" 1>&2
-            exit 1
-          fi
-        fi
-      fi
-    done
+echo '{ fetchNuGet }: ['
 
-    if [[ $found = false ]]; then
-      echo "couldn't find $package $version" >&2
-      exit 1
-    fi
+while [ "$index" -lt "${#depsParts[@]}" ]; do
+    pname="${depsParts[index]}"
+    version="${depsParts[++index]}"
+    sha256="${depsParts[++index]}"
+    hash="${depsParts[++index]}"
+    url="${depsParts[++index]}"
+    ((++index)) # Go to next pname
 
-    if [[ "$source" != https://api.nuget.org/v3/index.json ]]; then
-      echo "  (fetchNuGet { pname = \"$id\"; version = \"$version\"; hash = \"$hash\"; url = \"$url\"; })"
+    echo -n "  (fetchNuGet { pname = \"$pname\"; version = \"$version\"; "
+    if [ "$sha256" != null ]; then
+        echo -n "sha256 = \"$sha256\"; "
     else
-      echo "  (fetchNuGet { pname = \"$id\"; version = \"$version\"; hash = \"$hash\"; })"
+        echo -n "hash = \"$hash\"; "
+    fi
+    if [ "$url" != "null" ]; then
+        echo -n "url = \"$url\"; "
     fi
-  done
-  cd ..
+    echo '})'
 done
 
-cat << EOL
+cat <<EOL
 ]
 EOL
```

Comment:
Any reason to use `cat <<EOL` here instead of an `echo` or `echo -n`?

Reply:
No idea, I just didn't want to risk breaking compatibility.

Reply:
90% sure `echo ']'` does the same thing :thinking:

Reply:
Yeah, it does, just checked (was so paranoid I even used `hexdump` 😅). I'll update it.